controller/test/sensor_tests/sensors_test.cpp

```diff
@@ -72,42 +91,110 @@ TEST(SensorTests, FullScaleReading) {
 // If the Default PressureSensors::DEFAULT_VENTURI_PORT_DIAM and
 // DEFAULT_VENTURI_CHOKE_DIAM are changed from this, the tests will fail unless
 // you update the expected values accordingly.
+// Expected values from https://www.wolframalpha.com/input/?i=Venturi+flowmeter
 TEST(SensorTests, TestPositiveVolumetricFlowCalculation) {
   EXPECT_NEAR(Sensors::PressureDeltaToFlow(kPa(1.0f)).cubic_m_per_sec(),
-              9.52e-4f, COMPARISON_TOLERANCE_FLOW);
+              9.7162e-4f, COMPARISON_TOLERANCE_FLOW_CUBIC_M_PER_SEC);
   EXPECT_NEAR(Sensors::PressureDeltaToFlow(kPa(-1.0f)).cubic_m_per_sec(),
-              -9.52e-4f, COMPARISON_TOLERANCE_FLOW);
+              -9.7162e-4f, COMPARISON_TOLERANCE_FLOW_CUBIC_M_PER_SEC);
   EXPECT_NEAR(Sensors::PressureDeltaToFlow(kPa(0.0f)).cubic_m_per_sec(), 0.0f,
-              COMPARISON_TOLERANCE_FLOW);
+              COMPARISON_TOLERANCE_FLOW_CUBIC_M_PER_SEC);
   EXPECT_NEAR(Sensors::PressureDeltaToFlow(kPa(1.0e-7f)).cubic_m_per_sec(),
-              3.07e-7f, COMPARISON_TOLERANCE_FLOW);
+              3.0725e-7f, COMPARISON_TOLERANCE_FLOW_CUBIC_M_PER_SEC);
   EXPECT_NEAR(Sensors::PressureDeltaToFlow(kPa(100.0f)).cubic_m_per_sec(),
-              9.72e-3f, COMPARISON_TOLERANCE_FLOW);
+              9.7162e-3f, COMPARISON_TOLERANCE_FLOW_CUBIC_M_PER_SEC);
   EXPECT_NEAR(Sensors::PressureDeltaToFlow(kPa(-100.0f)).cubic_m_per_sec(),
-              -9.72e-3f, COMPARISON_TOLERANCE_FLOW);
+              -9.7162e-3f, COMPARISON_TOLERANCE_FLOW_CUBIC_M_PER_SEC);
+}
+
+TEST(SensorTests, TotalFlowCalculation) {
+  // These pressure waveforms start at 0 kPa to simulate the system being in the
+  // proper calibration state then they go over the sensor full ranges with
+  // less samples than for FullScaleReadings.
+  std::vector<Pressure> pressures = {kPa(0.0f), kPa(0.5f), kPa(1.0f), kPa(2.0f),
+                                     kPa(3.5f)};
+
+  // First set the simulated analog signals to an ambient 0 kPa corresponding
+  // voltage during calibration
+  Voltage voltage_at_0kPa = MPXV5004_PressureToVoltage(kPa(0)); //[V]
+  Hal.test_setAnalogPin(AnalogPin::PATIENT_PRESSURE, voltage_at_0kPa);
+  Hal.test_setAnalogPin(AnalogPin::INFLOW_PRESSURE_DIFF, voltage_at_0kPa);
+  Hal.test_setAnalogPin(AnalogPin::OUTFLOW_PRESSURE_DIFF, voltage_at_0kPa);
+
+  Sensors sensors;
+
+  for (auto p_in : pressures) {
+    for (auto p_out : pressures) {
+      auto readings =
+          update_readings(seconds(0.0f), kPa(0.0f), p_in, p_out, &sensors);
+
+      EXPECT_NEAR(ml_per_min(readings.flow_ml_per_min).cubic_m_per_sec(),
+                  (Sensors::PressureDeltaToFlow(p_in) -
+                   Sensors::PressureDeltaToFlow(p_out))
+                      .cubic_m_per_sec(),
+                  COMPARISON_TOLERANCE_FLOW_CUBIC_M_PER_SEC);
+    }
+  }
+}
+
+// start integrating from time=10 days to maximize potential floating
+// precision error during integration (which uses dt.minutes())
```

Comment:
But it is dt.minutes(), not t.minutes_since_startup(), right? We should never see a dt of more than a few milliseconds.

Reply:
You are correct... I'll go back to a more reasonable value to not raise more questions.